File: easy/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const mp = {}

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i]

    if (diff in mp) return [mp[diff], i]

    mp[nums[i]] = i
  }
}

let nums = [3, 3],
  target = 6

console.log(twoSum(nums, target))

File: easy/TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i, v in enumerate(nums):
            hold = target - v
            nums[i] = None
            if hold in nums:
                return [i, nums.index(hold)]
            nums[i] = v
        return []


nums = [5, 5]
target = 10


sol = Solution()
print(sol.twoSum(nums, target))
